pkg/operator/mon/health.go

```diff
@@ -95,7 +95,7 @@ func (c *Cluster) checkHealth() error {
 
 			// when the timeout for the mon has been reached, continue to the
 			// normal failover/delete mon pod part of the code
-			if time.Since(c.monTimeoutList[mon.Name]) > MonOutTimeout {
+			if time.Since(c.monTimeoutList[mon.Name]) <= MonOutTimeout {
```

Comment:
Ha, I looked at that condition yesterday and was wondering if it wasn't right. 
Did you get a chance to test this? Looks like we could use better instructions on how to test a private build.
https://github.com/rook/rook/blob/master/Documentation/development-flow.md

Reply:
I'm testing it on my cluster right now. Yes, some more documentation how to test locally with more than one node for testing the HA aspect of rook and Ceph would be nice.

Reply:
For this change, you can actually test with one node. If you delete the replicaset for one of the mons, the failover to a new pod will be done. For the multi-node setup locally, I'm using coreos-kubernetes: https://github.com/rook/coreos-kubernetes/tree/master. From the `multi-node/vagrant` directory, `vagrant up`.

Reply:
Thanks for the hint!
I have now tested it locally and it looks good. The log lines from the failover look as expected:
```
2017-09-08 15:28:17.140770 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:17.140802 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:27.353516 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:27.353660 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:37.564935 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:37.565122 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:47.783051 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:47.783090 I | op-mon: Failing over monitor rook-ceph-mon0
2017-09-08 15:28:47.783097 I | op-mon: starting new mon rook-ceph-mon3
2017-09-08 15:28:47.790047 I | op-mon: mon rook-ceph-mon3 running at 10.0.0.186:6790
```
<details>
<summary>Full rook-operator logs: Click to expand</summary>
<pre>
kubectl logs -f po/rook-operator-605083233-r7vdg
2017-09-08 15:27:07.944348 I | rook: starting Rook v0.5.0-68.gb3f365f.dirty with arguments '/usr/local/bin/rook operator'
2017-09-08 15:27:07.944615 I | rook: flag values: --help=false, --log-level=INFO, --mon-healthcheck-interval=10s, --mon-out-timeout=30s
2017-09-08 15:27:07.945614 I | rook: starting operator
2017-09-08 15:27:07.955348 I | op-kit: creating cluster CRD
2017-09-08 15:27:07.960454 I | op-kit: creating pool CRD
2017-09-08 15:27:09.004845 I | op-kit: start watching cluster resource in all namespaces at v1alpha1
2017-09-08 15:27:18.898968 I | op-cluster: starting cluster rook in namespace rook
2017-09-08 15:27:24.903516 I | op-mon: start running mons
2017-09-08 15:27:24.906149 I | op-mon: creating mon secrets for a new cluster
2017-09-08 15:27:24.906295 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook/mon.keyring --gen-key -n mon. --cap mon 'allow *'
2017-09-08 15:27:24.932132 I | exec: Running command: ceph-authtool --create-keyring /var/lib/rook/rook/client.admin.keyring --gen-key -n client.admin --set-uid=0 --cap mon 'allow *' --cap osd 'allow *' --cap mgr 'allow *' --cap mds 'allow'
2017-09-08 15:27:24.974431 I | op-mon: saved rook-admin secret
2017-09-08 15:27:24.980940 I | op-mon: saved mon endpoints to config map map[data: maxMonId:-1]
2017-09-08 15:27:24.981347 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:27:24.981437 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:27:24.981667 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:27:24.981747 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:27:24.987553 I | op-mon: mon rook-ceph-mon0 running at 10.0.0.178:6790
2017-09-08 15:27:25.006562 I | op-mon: saved mon endpoints to config map map[data:rook-ceph-mon0=10.0.0.178:6790 maxMonId:0]
2017-09-08 15:27:25.007044 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:27:25.007201 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:27:25.010578 I | op-mon: there are 1 nodes available for 1 mons
2017-09-08 15:27:25.013451 I | op-mon: Found 1 running nodes without mons
2017-09-08 15:27:25.021539 I | op-mon: mons created: 1
2017-09-08 15:27:25.021553 I | cephmon: waiting for mon quorum
2017-09-08 15:27:25.021612 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/838460308
2017-09-08 15:27:35.324101 I | cephmon: Ceph monitors formed quorum
2017-09-08 15:27:35.331733 I | op-mon: mon rook-ceph-mon0 running at 10.0.0.178:6790
2017-09-08 15:27:35.335945 I | op-mon: mon rook-ceph-mon1 running at 10.0.0.105:6790
2017-09-08 15:27:35.375946 I | op-mon: saved mon endpoints to config map map[data:rook-ceph-mon0=10.0.0.178:6790,rook-ceph-mon1=10.0.0.105:6790 maxMonId:1]
2017-09-08 15:27:35.376176 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:27:35.376312 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:27:35.379902 I | op-mon: there are 1 nodes available for 2 mons
2017-09-08 15:27:35.383574 I | op-mon: Found 0 running nodes without mons
2017-09-08 15:27:35.383587 I | op-mon: All nodes are running mons. Adding all 1 nodes to the availability.
2017-09-08 15:27:35.387567 I | op-mon: replicaset rook-ceph-mon0 already exists
2017-09-08 15:27:35.397362 I | op-mon: mons created: 2
2017-09-08 15:27:35.397376 I | cephmon: waiting for mon quorum
2017-09-08 15:27:35.397435 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/628679139
2017-09-08 15:27:35.834311 W | cephmon: failed to find initial monitor rook-ceph-mon1 in mon map
2017-09-08 15:27:40.834624 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/764481510
2017-09-08 15:27:43.638336 I | cephmon: Ceph monitors formed quorum
2017-09-08 15:27:43.647241 I | op-mon: mon rook-ceph-mon0 running at 10.0.0.178:6790
2017-09-08 15:27:43.656717 I | op-mon: mon rook-ceph-mon1 running at 10.0.0.105:6790
2017-09-08 15:27:43.662199 I | op-mon: mon rook-ceph-mon2 running at 10.0.0.254:6790
2017-09-08 15:27:43.685051 I | op-mon: saved mon endpoints to config map map[data:rook-ceph-mon1=10.0.0.105:6790,rook-ceph-mon2=10.0.0.254:6790,rook-ceph-mon0=10.0.0.178:6790 maxMonId:2]
2017-09-08 15:27:43.685253 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:27:43.685424 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:27:43.688561 I | op-mon: there are 1 nodes available for 3 mons
2017-09-08 15:27:43.692755 I | op-mon: Found 0 running nodes without mons
2017-09-08 15:27:43.692769 I | op-mon: All nodes are running mons. Adding all 1 nodes to the availability.
2017-09-08 15:27:43.704283 I | op-mon: replicaset rook-ceph-mon0 already exists
2017-09-08 15:27:43.711979 I | op-mon: replicaset rook-ceph-mon1 already exists
2017-09-08 15:27:43.731028 I | op-mon: mons created: 3
2017-09-08 15:27:43.731042 I | cephmon: waiting for mon quorum
2017-09-08 15:27:43.731099 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/242781197
2017-09-08 15:27:44.068783 W | cephmon: failed to find initial monitor rook-ceph-mon2 in mon map
2017-09-08 15:27:49.069194 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/121494536
2017-09-08 15:27:49.832982 W | cephmon: initial monitor rook-ceph-mon2 is not in quorum list
2017-09-08 15:27:54.833579 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/919375303
2017-09-08 15:27:55.096689 I | cephmon: Ceph monitors formed quorum
2017-09-08 15:27:55.099741 I | op-cluster: creating initial crushmap
2017-09-08 15:27:55.099770 I | cephclient: setting crush tunables to firefly
2017-09-08 15:27:55.099851 I | exec: Running command: ceph osd crush tunables firefly --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format plain --out-file /tmp/407262330
2017-09-08 15:27:55.372728 I | exec: adjusted tunables profile to firefly
2017-09-08 15:27:55.372833 I | cephclient: succeeded setting crush tunables to profile firefly: 
2017-09-08 15:27:55.372926 I | exec: Running command: crushtool -c /tmp/094150545 -o /tmp/183123900
2017-09-08 15:27:55.391299 I | exec: Running command: ceph osd setcrushmap -i /tmp/183123900 --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/868588523
2017-09-08 15:27:56.415635 I | exec: 3
2017-09-08 15:27:56.415838 I | op-cluster: created initial crushmap
2017-09-08 15:27:56.420251 I | op-mgr: start running mgr
2017-09-08 15:27:56.422864 I | exec: Running command: ceph auth get-or-create-key mgr.rook-ceph-mgr0 mon allow * --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/549069390
2017-09-08 15:27:56.666413 I | op-mgr: rook-ceph-mgr0 deployment started
2017-09-08 15:27:56.666429 I | op-api: starting the Rook api
2017-09-08 15:27:56.715149 I | op-api: API service running at 10.0.0.179:8124
2017-09-08 15:27:56.880270 I | op-api: creating cluster role rook-api
2017-09-08 15:27:56.931102 I | op-api: api deployment started
2017-09-08 15:27:56.931117 I | op-osd: start running osds in namespace rook
2017-09-08 15:27:56.946315 I | op-osd: osd daemon set started
2017-09-08 15:27:56.946382 I | exec: Running command: ceph auth get-or-create-key client.rook-rook-user osd allow rwx mon allow r --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/973010773
2017-09-08 15:27:57.274187 I | op-cluster: saved rook-rook-user secret
2017-09-08 15:27:57.274200 I | op-api: retrieving rook api endpoint for namespace rook
2017-09-08 15:27:57.276750 I | op-api: rook api endpoint 10.0.0.179:8124 for namespace rook
2017-09-08 15:27:57.276764 I | op-cluster: Done creating rook instance in namespace rook
2017-09-08 15:27:57.277045 I | op-api: retrieving rook api endpoint for namespace rook
2017-09-08 15:27:57.282299 I | op-api: rook api endpoint 10.0.0.179:8124 for namespace rook
2017-09-08 15:27:57.282324 I | op-kit: start watching pool resource in namespace rook at v1alpha1
2017-09-08 15:28:17.140770 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:17.140802 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:27.353516 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:27.353660 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:37.564935 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:37.565122 W | op-mon: mon rook-ceph-mon0 NOT found in quorum, STILL in mon out timeout
2017-09-08 15:28:47.783051 W | op-mon: mon rook-ceph-mon0 NOT found in quorum. {Quorum:[0 2] MonMap:{Mons:[{Name:rook-ceph-mon1 Rank:0 Address:10.0.0.105:6790/0} {Name:rook-ceph-mon0 Rank:1 Address:10.0.0.178:6790/0} {Name:rook-ceph-mon2 Rank:2 Address:10.0.0.254:6790/0}]}}
2017-09-08 15:28:47.783090 I | op-mon: Failing over monitor rook-ceph-mon0
2017-09-08 15:28:47.783097 I | op-mon: starting new mon rook-ceph-mon3
2017-09-08 15:28:47.790047 I | op-mon: mon rook-ceph-mon3 running at 10.0.0.186:6790
2017-09-08 15:28:47.829093 I | op-mon: saved mon endpoints to config map map[data:rook-ceph-mon0=10.0.0.178:6790,rook-ceph-mon1=10.0.0.105:6790,rook-ceph-mon2=10.0.0.254:6790,rook-ceph-mon3=10.0.0.186:6790 maxMonId:2]
2017-09-08 15:28:47.829296 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:28:47.829433 I | cephmon: generated admin config in /var/lib/rook/rook
2017-09-08 15:28:47.841158 I | op-mon: there are 1 nodes available for 4 mons
2017-09-08 15:28:47.845215 I | op-mon: Found 0 running nodes without mons
2017-09-08 15:28:47.845227 I | op-mon: All nodes are running mons. Adding all 1 nodes to the availability.
2017-09-08 15:28:47.859655 I | op-mon: mons created: 1
2017-09-08 15:28:47.859680 I | cephmon: waiting for mon quorum
2017-09-08 15:28:47.859735 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/442111716
2017-09-08 15:28:48.171619 W | cephmon: failed to find initial monitor rook-ceph-mon3 in mon map
2017-09-08 15:28:53.171980 I | exec: Running command: ceph mon_status --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/400549107
2017-09-08 15:29:03.467697 I | cephmon: Ceph monitors formed quorum
2017-09-08 15:29:03.467749 I | op-mon: ensuring removal of unhealthy monitor rook-ceph-mon0
2017-09-08 15:29:03.471064 I | op-mon: dead mon rook-ceph-mon0 was already gone
2017-09-08 15:29:03.471315 I | exec: Running command: ceph mon remove rook-ceph-mon0 --cluster=rook --conf=/var/lib/rook/rook/rook.config --keyring=/var/lib/rook/rook/client.admin.keyring --format json --out-file /tmp/732007350
2017-09-08 15:29:03.702579 I | exec: removing mon.rook-ceph-mon0 at 10.0.0.178:6790/0, there will be 3 monitors
2017-09-08 15:29:03.702770 I | cephmon: removed monitor rook-ceph-mon0
2017-09-08 15:29:03.748218 I | op-mon: saved mon endpoints to config map map[data:rook-ceph-mon1=10.0.0.105:6790,rook-ceph-mon2=10.0.0.254:6790,rook-ceph-mon3=10.0.0.186:6790 maxMonId:3]
2017-09-08 15:29:03.748455 I | cephmon: writing config file /var/lib/rook/rook/rook.config
2017-09-08 15:29:03.748577 I | cephmon: generated admin config in /var/lib/rook/rook
</pre>
</details>

Reply:
Looks great, i'll merge it